### ISSUE TITLE
Can't process audio files longer than a minute

### ISSUE THREAD

**Issue by GiraFarkas** [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#2394]
When trying to process audio source files longer than a minute for Inference the webui just states 'Error' and there's no log in the cmd at all. It's the same for downloaded as well as for trained models. Strangely, sometimes the same file wont work that could be processed last time and vice versa. Shorten the files and converting to wav seems to increase chances of success but I did not find any 100% reproducable pattern so far. In addition, processing these files is possible when using batch mode and putting a file that probably works (a wav with < 1 min) at the top of the list. Anyone has an idea what's going on there? Thanks!

Running in an virtual environment on Windows, here's the pip list.

(myenv) C:\RVC-Project\Retrieval-based-Voice-Conversion-WebUI-main>pip list
Package                   Version
------------------------- ------------
absl-py                   2.1.0
aiofiles                  24.1.0
aiohappyeyeballs          2.4.3
aiohttp                   3.11.7
aiosignal                 1.3.1
altair                    5.5.0
antlr4-python3-runtime    4.8
anyio                     4.6.2.post1
aria2                     0.0.1b0
async-timeout             5.0.1
attrs                     24.2.0
audioread                 3.0.1
av                        13.1.0
bitarray                  3.0.0
certifi                   2024.8.30
cffi                      1.17.1
charset-normalizer        3.4.0
click                     8.1.7
colorama                  0.4.6
coloredlogs               15.0.1
contourpy                 1.3.1
cycler                    0.12.1
Cython                    3.0.11
decorator                 5.1.1
einops                    0.8.0
exceptiongroup            1.2.2
fairseq                   0.12.2
faiss-cpu                 1.7.3
fastapi                   0.88.0
ffmpeg-python             0.2.0
ffmpy                     0.3.1
filelock                  3.16.1
flatbuffers               24.3.25
fonttools                 4.55.0
frozenlist                1.5.0
fsspec                    2024.10.0
future                    1.0.0
gradio                    3.34.0
gradio_client             1.4.3
grpcio                    1.68.0
h11                       0.14.0
httpcore                  1.0.7
httpx                     0.27.2
huggingface-hub           0.26.2
humanfriendly             10.0
hydra-core                1.0.7
idna                      3.10
Jinja2                    3.1.4
joblib                    1.4.2
json5                     0.9.28
jsonschema                4.23.0
jsonschema-specifications 2024.10.1
kiwisolver                1.4.7
librosa                   0.9.1
linkify-it-py             2.0.3
llvmlite                  0.39.0
local-attention           1.9.15
lxml                      5.3.0
Markdown                  3.7
markdown-it-py            2.2.0
MarkupSafe                3.0.2
matplotlib                3.9.2
matplotlib-inline         0.1.7
mdit-py-plugins           0.3.3
mdurl                     0.1.2
mpmath                    1.3.0
multidict                 6.1.0
narwhals                  1.14.2
networkx                  3.4.2
numba                     0.56.4
numpy                     1.23.5
omegaconf                 2.0.6
onnxruntime-gpu           1.20.1
orjson                    3.10.12
packaging                 24.2
pandas                    2.2.3
pillow                    11.0.0
pip                       22.2.1
platformdirs              4.3.6
pooch                     1.8.2
portalocker               3.0.0
praat-parselmouth         0.4.5
propcache                 0.2.0
protobuf                  5.28.3
pyasn1                    0.6.1
pyasn1_modules            0.4.1
pycparser                 2.22
pydantic                  1.10.19
pydub                     0.25.1
Pygments                  2.18.0
pyparsing                 3.2.0
pyreadline3               3.5.4
python-dateutil           2.9.0.post0
python-dotenv             1.0.1
python-multipart          0.0.17
pytz                      2024.2
pywin32                   308
pyworld                   0.3.2
PyYAML                    6.0.2
referencing               0.35.1
regex                     2024.11.6
requests                  2.32.3
resampy                   0.4.3
rpds-py                   0.21.0
sacrebleu                 2.4.3
scikit-learn              1.5.2
scipy                     1.14.1
semantic-version          2.10.0
setuptools                63.2.0
six                       1.16.0
sniffio                   1.3.1
soundfile                 0.12.1
starlette                 0.22.0
sympy                     1.13.1
tabulate                  0.9.0
tensorboard               2.18.0
tensorboard-data-server   0.7.2
tensorboardX              2.6.2.2
threadpoolctl             3.5.0
torch                     2.0.1+cu117
torchaudio                2.0.2+cu117
torchcrepe                0.0.20
torchfcpe                 0.0.4
torchvision               0.15.2+cu117
tornado                   6.4.2
tqdm                      4.67.1
traitlets                 5.14.3
typing_extensions         4.12.2
tzdata                    2024.2
uc-micro-py               1.0.3
urllib3                   2.2.3
uvicorn                   0.32.1
websockets                12.0
Werkzeug                  3.1.3
yarl                      1.18.0

**Comment by tanlianhui** [comment]
same issue; still looking for an answer

**Comment by LOOKrecy** [comment]
I encountered the exact same issue. After a series of experiments, I discovered that the problem is not with the model or the processing methods but solely with how Gradio, which runs the web server, functions. It simply doesn’t wait for the processing to complete and throws an error. If you’ve noticed, the error only occurs in the WebUI, while the console remains error-free.

Additionally, I added debug messages to various functions (e.g., _vc_single_ in _modules.py_) and found that everything completes successfully and works perfectly, even if the WebUI shows an "Error" on the page.

There are also no issues with file length when using the CLI instead of the WebGUI. Files of any length are converted without problems. 

You can try using the CLI yourself:  
In the command line, enter something like*:  
`python tools\infer_cli.py --input_path "YOUR_PATH\input.wav" --index_path "YOUR_PATH\INDEX_YOUR_MODEL.index" --f0method rmvpe --opt_path "YOUR_PATH\output.wav" --model_name "YOUR_MODEL.pth" --device cuda:0 --is_half True`

*The arguments may vary, especially those related to CUDA usage.

**Comment by tanlianhui** [comment]
Thanks! It works! Any idea why Gradio has this issue though?

**Comment by LOOKrecy** [comment]
Good news, guys, **I think I’ve found the cause of the issue and a temporary solution**!  

The problem seems to lie in the updated **httpx** module, which is used either by Gradio or the project itself. This module is listed in **requirements.txt**, but there are no version constraints specified for it. For example, I have version 0.28.1 installed.  

Basically, the module sets a default timeout for data requests to **5 seconds**. Ideally, the calling method is supposed to override this value, but apparently, the RVC author didn’t do that for some reason. This is why the connection drops before the processing is complete, even though the processing itself continues and finishes successfully.  

Ideally, the timeout value should be redefined somewhere in the code, but that’s something the author needs to address. I’ll dig around and try to find the correct approach, but for now, I’ll offer you a workaround.  

### Steps for the workaround:
1. **Find where your httpx package is installed.**  
   You can do this by running the following command in the terminal:  
   ```
   pip show httpx
   ```  
   Depending on whether you’re using a local or global environment, the paths might differ. In a local environment, the path might look like this: `venv\Lib\site-packages\httpx`

2. **Navigate to the folder containing httpx.**  
   Once inside, locate the file named **_config.py**.

3. **Modify the timeout value.**  
   At the very end of the **_config.py** file, you’ll find the following line:  
   ```
   DEFAULT_TIMEOUT_CONFIG = Timeout(timeout=5.0)
   ```  
   Change the value **5.0** to the desired timeout in seconds, save the changes, and restart the server.

This should fix the issue!

**Comment by weaponmasterjax** [comment]
> Good news, guys, **I think I’ve found the cause of the issue and a temporary solution**!
> 
> The problem seems to lie in the updated **httpx** module, which is used either by Gradio or the project itself. This module is listed in **requirements.txt**, but there are no version constraints specified for it. For example, I have version 0.28.1 installed.
> 
> Basically, the module sets a default timeout for data requests to **5 seconds**. Ideally, the calling method is supposed to override this value, but apparently, the RVC author didn’t do that for some reason. This is why the connection drops before the processing is complete, even though the processing itself continues and finishes successfully.
> 
> Ideally, the timeout value should be redefined somewhere in the code, but that’s something the author needs to address. I’ll dig around and try to find the correct approach, but for now, I’ll offer you a workaround.
> 
> ### Steps for the workaround:
> 1. **Find where your httpx package is installed.**
>    You can do this by running the following command in the terminal:
>    ```
>    pip show httpx
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    Depending on whether you’re using a local or global environment, the paths might differ. In a local environment, the path might look like this: `venv\Lib\site-packages\httpx`
> 2. **Navigate to the folder containing httpx.**
>    Once inside, locate the file named **_config.py**.
> 3. **Modify the timeout value.**
>    At the very end of the **_config.py** file, you’ll find the following line:
>    ```
>    DEFAULT_TIMEOUT_CONFIG = Timeout(timeout=5.0)
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    Change the value **5.0** to the desired timeout in seconds, save the changes, and restart the server.
> 
> This should fix the issue!

Nice solution! Works flawlessly.
